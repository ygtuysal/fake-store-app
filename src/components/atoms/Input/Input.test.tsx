import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { ThemeProvider } from 'styled-components';
import { Input } from '.';

const theme = {
  colors: {
    primary: '#0070f3',
    text: '#111',
    error: '#dc3545',
    gray: {
      100: '#f1f3f5',
      400: '#ced4da',
    },
    border: '#dee2e6',
    background: '#ffffff',
  },
  borderRadius: {
    md: '4px',
  },
  transitions: {
    fast: '0.2s',
  },
  spacing: {
    xs: '4px',
  },
};

const renderWithTheme = (ui: React.ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);

describe('Input', () => {
  it('renders default input', () => {
    const { container } = renderWithTheme(<Input placeholder="type here" />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border', `2px solid ${theme.colors.border}`);
  });

  it('renders full width input', () => {
    const { container } = renderWithTheme(<Input $fullWidth />);
    expect(container.firstChild).toHaveStyleRule('width', '100%');
  });

  it('renders input with error', () => {
    const { container } = renderWithTheme(<Input $error />);
    expect(container.firstChild).toHaveStyleRule('border', `2px solid ${theme.colors.error}`);
  });

  it('renders disabled input', () => {
    const { container } = renderWithTheme(<Input disabled />);
    expect(container.firstChild).toHaveStyleRule('cursor', 'not-allowed', {
      modifier: ':disabled',
    });
  });

  it('renders focus style', () => {
    const { container } = renderWithTheme(<Input />);
    expect(container.firstChild).toHaveStyleRule('box-shadow', `0 0 0 3px ${theme.colors.primary}33`, {
      modifier: ':focus',
    });
  });
});
