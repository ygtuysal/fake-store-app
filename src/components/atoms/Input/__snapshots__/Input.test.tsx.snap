// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Input renders default input 1`] = `
.c0 {
  width: auto;
  padding: 10px 16px;
  font-size: 16px;
  border: 2px solid #dee2e6;
  border-radius: 4px;
  background-color: #ffffff;
  color: #111;
  transition: all 0.2s;
  outline: none;
}

.c0:focus {
  border-color: #0070f3;
  box-shadow: 0 0 0 3px #0070f333;
}

.c0:hover:not(:disabled) {
  border-color: #ced4da;
}

.c0:disabled {
  background-color: #f1f3f5;
  cursor: not-allowed;
  opacity: 0.6;
}

.c0::placeholder {
  color: #ced4da;
}

<input
  class="c0"
  placeholder="type here"
/>
`;
